tests/core/test_agent.py

```diff
@@ -194,6 +197,37 @@ def test_agent_wrong_use_of_load(tmpdir, default_domain):
     assert agent.model_directory is not None
 
 
+def test_init_agent_with_missing_policies():
+    TRIGGERS_AND_FORMS_DOMAIN = {
+        "intents": [{"affirm": {"triggers": "utter_ask_num_people"}}],
+        "templates": {"utter_ask_num_people": [{"text": "how many people?"}]},
+        "actions": ["utter_ask_num_people"],
+        "forms": ["restaurant_form"],
+    }
+
+    NO_MAPPING_POLICY_CONFIG = {
+        "policies": [{"name": "MemoizationPolicy"}, {"name": "FormPolicy"}]
+    }
+
+    NO_FORMS_POLICY_CONFIG = {
+        "policies": [{"name": "MemoizationPolicy"}, {"name": "MappingPolicy"}]
+    }
+
+    with pytest.raises(InvalidDomain) as execinfo:
```

Comment:
how about the parametrize stuff?

Reply:
We discussed it's going to be either defining it inside here or parameterizing it, right?

Reply:
Ah, then I got you wrong. I'd actually put it in `pytest.mark.parametrize` (I also think the exception messages does not have to be tested)

Reply:
But what are we parametrizing here?

Reply:
I thought about doing something like (also minimizing the domain to the minimal required case):

```python
@pytest.mark.parametrize(
    "domain, policy_config",
    [
        (
            {"forms": ["restaurant_form"]},
            {"policies": [{"name": "MemoizationPolicy"}, {"name": "MappingPolicy"}]},
        ),
        (
            {"intents": [{"affirm": {"triggers": "utter_ask_num_people"}}]},
            {"policies": [{"name": "MemoizationPolicy"}, {"name": "FormPolicy"}]},
        ),
    ],
)
def test_incompatible_domain_and_policy(domain: Dict, policy_config: Dict):
```

Reply:
I think it might be better to split up these tests. While parametrize cleans it up very well, I'm afraid it sacrifices readability as it's hard to understand what we're testing for without the presence of the exception messages asserts.

Reply:
cool with me!